ssl/statem/statem_lib.c

```diff
@@ -1956,6 +1958,8 @@ int ssl_choose_client_version(SSL *s, int version, RAW_EXTENSION *extensions)
  * @s: The SSL connection
  * @min_version: The minimum supported version
  * @max_version: The maximum supported version
+ * @real_max:    The highest version below the lowest compile time version hole
+ *               that lies above at least one run-time enabled protocol.
```

Comment:
Again, likely sufficient, but perhaps a bit ambiguous, it is the hole and not `real_max` that lies above at least one run-time enabled protocol. The latter may be equal to rather than lie above that run-time enabled protocol. Slightly more verbose comments are good where there's complicated intent.